compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -34,4 +40,110 @@ luci::CircleNode *CloneNodeLet<CN::OPQR>::visit(const luci::CircleReshape *node)
   return cloned;
 }
 
+namespace sinf
+{
+
+/**
+ * @note  CircleReshape has new shape info in two places: 2nd input and attribute.
+ *        This shape inference uses shape from input 'shape' node when it's constant.
+ *        If not, shape will be from node itself. shape from attribute is not used.
+ *
+ * TODO Change this policy when not appropriate
+ */
+loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
+{
+  LOGGER(l);
+
+  const loco::DataType S32 = loco::DataType::S32;
+
+  bool is_input_known = true;
+  auto input_tensor = dynamic_cast<luci::CircleNode *>(node->tensor());
+  for (uint32_t axis = 0; axis < input_tensor->rank(); ++axis)
+  {
+    if (!input_tensor->dim(axis).known())
+    {
+      is_input_known = false;
+      break;
+    }
+  }
+
+  loco::TensorShape shape_by_input;
+  {
+    LUCI_ASSERT(node->shape(), "2nd input shape() should not be nullptr");
+
+    // Only support node's shape() is CircleConst with S32
+    // TODO support other node with other types
+    auto const_shape_node = dynamic_cast<luci::CircleConst *>(node->shape());
+    if (const_shape_node != nullptr)
+    {
+      LUCI_ASSERT(const_shape_node->dtype() == S32, "Only support int32 CircleConst");
+
+      shape_by_input.rank(const_shape_node->size<S32>());
+
+      for (uint32_t axis = 0; axis < shape_by_input.rank(); ++axis)
+      {
+        shape_by_input.dim(axis) = const_shape_node->at<S32>(axis);
+        if (const_shape_node->at<S32>(axis) < 0)
+          shape_by_input.dim(axis).unset();
+      }
+    }
+    else
+    {
+      // We use shape from the node itself
+      shape_by_input = own_shape(node);
```

Comment:
As I said in https://github.com/Samsung/ONE/pull/13866#discussion_r1739818477, these lines should be substitued by `shape_by_attr` logic.

Reply:
Thank you so much for your kind and insightful reviews.
I’ve carefully read through all of them and applied the suggested changes, except for the `own_shape` one.
However, I now have a complete understanding of the `own_shape` method, and I will work on replacing it with new logic (including updating the recipe).

As I mentioned in https://github.com/Samsung/ONE/pull/13912#discussion_r1741534197, I would like to close this PR and start over to ensure a more readable and understandable commit.

I sincerely hope you continue to take an interest in this issue and offer your support. 
Thanks again!

Reply:
Thank you for the kind response and sorry(?) for your effort :)
After creating new PR, please register me as a reviewer!
I will gladly take an interest for that :D